src/Features/Core/Portable/EditAndContinue/Remote/RemoteEditAndContinueServiceProxy.cs

```diff
@@ -130,25 +131,28 @@ private IEditAndContinueService GetLocalService()
         if (client == null)
         {
             var sessionId = await GetLocalService().StartDebuggingSessionAsync(solution, debuggerService, sourceTextProvider, captureMatchingDocuments, captureAllMatchingDocuments, reportDiagnostics, cancellationToken).ConfigureAwait(false);
-            return new RemoteDebuggingSessionProxy(solution.Services, LocalConnection.Instance, sessionId);
+            using var disposable = new ReferenceCountedDisposable<IDisposable>(LocalConnection.Instance);
+            // RemoteDebuggingSessionProxy will add its own refcount to 'disposable'.
+            return new RemoteDebuggingSessionProxy(solution.Services, disposable, sessionId);
         }
 
         // need to keep the providers alive until the session ends:
-        var connection = client.CreateConnection<IRemoteEditAndContinueService>(
-            callbackTarget: new DebuggingSessionCallback(debuggerService, sourceTextProvider));
+        //
+        // Wrap with a ref-counted-disposable here to ensure we clean this up properly if we don't transfer ownership to the RemoteDebuggingSessionProxy.
+        using var connection = new ReferenceCountedDisposable<RemoteServiceConnection<IRemoteEditAndContinueService>>(
```

Comment:
ref count of connection will start at 1.

Reply:
this code ensures that if we don't properly do all the following steps that we dispose of hte connection.  ew don't want connections leaked if we move to a world where we hold off on disposing the RemoteHostClient while there are still connections open to the OOP server (see https://github.com/dotnet/roslyn/pull/73467)